### PR TITLE
[AKS] add support to create cluster with cluster snapshot

### PR DESCRIPTION
**Pull request by zqingqing1**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/HISTORY.md [hunk 1]
```diff
@@ -233,7 +239,7 @@ and `az aks nodepool add`
 0.5.24
 +++++
 
-* * Add "--aks-custom-headers" for "az aks nodepool upgrade"
+  * * Add "--aks-custom-headers" for "az aks nodepool upgrade"
```
Comment (zhoxing-ms) [inline src/aks-preview/HISTORY.md:242]: Why add a space here?

File: src/aks-preview/azext_aks_preview/_params.py [hunk 2]
```diff
@@ -202,6 +202,8 @@ def load_arguments(self, _):
         c.argument('workload_runtime', arg_type=get_enum_type(
             workload_runtimes), default=CONST_WORKLOAD_RUNTIME_OCI_CONTAINER)
         c.argument('snapshot_id', type=str, validator=validate_snapshot_id)
+        c.argument('cluster_snapshot_id', type=str,
+                   validator=validate_cluster_snapshot_id, is_preview=True)
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_params.py:206]: > type=str

The default value of `type` is `str`, so there is no need to add extra declarations

**Comment by zhoxing-ms** [comment]
@zqingqing1 Could you please address those CI issues~

File: src/aks-preview/HISTORY.md (hunk 1 above)
Comment (zqingqing1) [reply src/aks-preview/HISTORY.md:242]: added by mistake. will remove it

File: src/aks-preview/azext_aks_preview/_params.py (hunk 2 above)
Comment (zqingqing1) [reply src/aks-preview/azext_aks_preview/_params.py:206]: Ok, removed it

File: src/aks-preview/HISTORY.md [hunk 3]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+0.5.61
```
[the remark this answers is not in the document]
Comment (zqingqing1) [reply src/aks-preview/HISTORY.md:6]: we can resolve conflict when needed

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**